src/main/java/com/hongliangjie/fugue/topicmodeling/LDA/LDA.java

```diff
@@ -378,31 +392,94 @@ else if ("binary".equals(samplerStr)){
-            saveModel();
+            saveModel(0);
     }
 
     public void test() {
+        try {
+            loadModel();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
 
+    public void loadModel() throws IOException {
+        int multipleModels = Integer.parseInt(cmdArg.getParam("multipleModels").toString());
+        String[] modelFileNames = null;
+        if (multipleModels == 1) {
+            modelFileNames = getModelFiles(true);
+        }
+        else{
+            modelFileNames = new String[1];
+            modelFileNames[0] = cmdArg.getParam("modelFile").toString();
+        }
+        Gson gson = new Gson();
+        String line;
+        modelPools = new ArrayList<ModelCountainer>();
+        for (int s = 0; s < modelFileNames.length; s++) {
+            String modelFileName = modelFileNames[s];
+            File modelFile = new File(modelFileName);
+            if (modelFile.exists() && !modelFile.isDirectory()) {
+                BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(modelFile), "UTF8"));
+                while ((line = br.readLine()) != null) {
+                    Model obj = gson.fromJson(line, LDAModel.class);
+                    Message msg = obj.getParameters();
+                    ModelCountainer currentModel = new ModelCountainer();
+                    currentModel.alpha = (double[])msg.getParam("alpha");
+                    currentModel.beta = (List<Double>)msg.getParam("beta");
+                    currentModel.outsideWordTopicCounts = (Map<String, int[]>)msg.getParam("wordTopicCounts");
+                    currentModel.topicCounts = (int[])msg.getParam("topicCounts");
+                    // recompute alphaSum and betaSum
+                    currentModel.alphaSum = 0.0;
+                    for (int k = 0; k < currentModel.alpha.length; k++)
+                        currentModel.alphaSum += currentModel.alpha[k];
+                    currentModel.betaSum = 0.0;
+                    for (int v = 0; v < currentModel.beta.size(); v++)
+                        currentModel.betaSum += currentModel.beta.get(v);
+                    modelPools.add(currentModel);
+                    break;
+                }
+                LOGGER.info("Loaded " + modelFileName);
+            }
+        }
     }
 
-    public void saveModel() {
+    private String[] getModelFiles(boolean all){
         String[] outputFileParts = cmdArg.getParam("modelFile").toString().split(Pattern.quote("."));
         StringBuilder outputFilePrefix = new StringBuilder();
         for(int i = 0; i < outputFileParts.length - 1; i ++){
             outputFilePrefix.append(outputFileParts[i] + ".");
         }
-        outputFilePrefix.append(Integer.toString(SAVED % 10) + ".");
-        outputFilePrefix.append(outputFileParts[outputFileParts.length - 1]);
-        String outputFileName = outputFilePrefix.toString();
+        if (all == false) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary comparisons in boolean expressions](https://www.codacy.com/app/hongliangjie/Fugue/file/2253567181/issues/source?bid=3162994&fileBranchId=3175173#l455)